.github/scripts/build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -eux
+
+. .github/scripts/env.sh
+. .github/scripts/common.sh
+
+uname -a
+uname -p
+uname
+pwd
+env
+
+# ensure ghcup
+if ! command -v ghcup ; then
+	install_ghcup
+fi
+
+# ensure cabal-cache
+download_cabal_cache "$HOME/.local/bin/cabal-cache"
+
+
+# build
+ecabal update
+ghcup install ghc "${GHC_VERSION}"
+ghcup set ghc "${GHC_VERSION}"
+"ghc-${GHC_VERSION}" --info
+"ghc" --info
+
+mkdir -p "$CI_PROJECT_DIR/out/${ARTIFACT}"
+mkdir -p "$CI_PROJECT_DIR/out/plan.json"
+
+case "$(uname)" in
+    MSYS_*|MINGW*)
+		args=( -O2 -w "ghc-$GHC_VERSION" --project-file cabal.project --disable-profiling --disable-tests --enable-executable-stripping ${ADD_CABAL_ARGS})
```

Comment:
Doesn't `ghcup set ghc` mean we don't need `-w`? `--project-file` also seems redundant.

For the others, I again wonder whether it would be simpler to setup a `cabal.project.local` in a standalone step. That's what we do currently, and it makes it somewhat easier to see the different settings we're using for different platforms (we could even make it a build artifact!).